### DISCUSSION TITLE
Building the project for Gamecube internal BlueRetro adapter

### DISCUSSION THREAD

**Discussion by coolbho3k** [darthcloud/BlueRetro#486] [General]
Hey all, I just got my Gamecube internal BlueRetro adapter and want to make some contributions to this project with it. :) 

Here is my setup:

1. Follow [these instructions](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/linux-macos-setup.html) to get ESP-IDF.

2. Clone the BlueRetro repo and also init/update the submodules.

3. `idf.py set-target esp32`

4. `idf.py menuconfig`

I set the configuration: hw2, gc. I'm leaving everything else on default!

5. `idf.py build`

I'm worried that this will brick my adapter if I've done anything wrong. Would the resulting `build/BlueRetro.bin` be safe to flash via the web interface?

**Comment by darthcloud** [comment]
Proper way to setup BlueRetro env is described here:
https://github.com/darthcloud/BlueRetroRoot

Not sure if you got the right idf version the way you did it.

**Comment by coolbho3k** [comment]
All right, I'll use the IDF version there and follow those instructions. Thanks! There's not a big risk of bricking it right? Not sure if there's a really easy way to connect to USB to the Laser Bear board. 

Actually, on second thought, I might just order an extra ESP32 board for helping out with development… Sounds a bit easier.

**Reply by darthcloud** [reply]
Yeah get a separate ESP32, OTA update is very long so GC internal is not easiest dev setup.

You can always program via a special cable with 4 GC plug, not sure is lazer bear got that available yet.
